lib/_http_server.js

```diff
@@ -286,8 +286,20 @@ function Server(options, requestListener) {
     throw new ERR_INVALID_ARG_TYPE('options', 'object', options);
   }
 
-  this[kIncomingMessage] = options.IncomingMessage || IncomingMessage;
-  this[kServerResponse] = options.ServerResponse || ServerResponse;
+  Object.defineProperties(this, {
```

Comment:
Maybe it would be worthwhile to put these into a helper and then document about `util.inspect.defaultOptions` in the definition of that helper because to understand what those properties are one has to look into the source code anyway

Reply:
Shall I do this in this specific PR or is that fine for a separate one?

Reply:
I think it should be in this PR.